Game/RainMeadow.PlayerHooks.cs

```diff
@@ -190,52 +192,81 @@ private void Player_GrabUpdate(ILContext il)
         }
     }
 
+    public static bool sUpdateFood = true;
+
     private void Player_AddQuarterFood(On.Player.orig_AddQuarterFood orig, Player self)
     {
+        var state = (PlayerState)self.State;
+        var origFood = state.foodInStomach * 4 + state.quarterFoodPoints;
         if (OnlineManager.lobby != null)
         {
             if (!OnlinePhysicalObject.map.TryGetValue(self.abstractPhysicalObject, out var onlineEntity)) throw new InvalidProgrammerException("Player doesn't have OnlineEntity counterpart!!");
             if (!onlineEntity.isMine) return;
+        }
 
+        var origUpdateFood = sUpdateFood;
+        sUpdateFood = false;
+        orig(self);
+        sUpdateFood = origUpdateFood;
```

Comment:
this is needed because AddQuarterFood calls AddFood when overflow
and it felt weird to special-case it for just AddQuarterFood
so i added this for all the Food functions